app/src/main/kotlin/com/wire/android/ui/common/banner/SecurityClassificationViewModel.kt

```diff
@@ -0,0 +1,81 @@
+/*
+ * Wire
+ * Copyright (C) 2023 Wire Swiss GmbH
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program. If not, see http://www.gnu.org/licenses/.
+ */
+
+package com.wire.android.ui.common.banner
+
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.setValue
+import androidx.lifecycle.SavedStateHandle
+import androidx.lifecycle.ViewModel
+import androidx.lifecycle.viewModelScope
+import com.wire.android.appLogger
+import com.wire.android.di.scopedArgs
+import com.wire.kalium.logic.data.id.ConversationId
+import com.wire.kalium.logic.data.id.QualifiedID
+import com.wire.kalium.logic.data.user.UserId
+import com.wire.kalium.logic.feature.conversation.GetOtherUserSecurityClassificationLabelUseCase
+import com.wire.kalium.logic.feature.conversation.ObserveSecurityClassificationLabelUseCase
+import com.wire.kalium.logic.feature.conversation.SecurityClassificationType
+import dagger.hilt.android.lifecycle.HiltViewModel
+import kotlinx.coroutines.launch
+import javax.inject.Inject
+
+interface SecurityClassificationViewModel {
+    fun state(): SecurityClassificationType
+}
+
+@HiltViewModel
+class SecurityClassificationViewModelImpl @Inject constructor(
+    private val observeSecurityClassificationLabel: ObserveSecurityClassificationLabelUseCase,
+    private val getOtherUserSecurityClassificationLabel: GetOtherUserSecurityClassificationLabelUseCase,
+    savedStateHandle: SavedStateHandle
+) : SecurityClassificationViewModel, ViewModel() {
+
+    private val args: SecurityClassificationArgs = savedStateHandle.scopedArgs()
+    private val conversationId: QualifiedID? = args.conversationId
+    private val userId: QualifiedID? = args.userId
+
+    private var state by mutableStateOf(SecurityClassificationType.NONE)
+
+    override fun state(): SecurityClassificationType = state
+
+    init {
+        when {
+            conversationId != null -> fetchConversationClassificationType(conversationId)
+            userId != null -> fetchUserClassificationType(userId)
+            else -> appLogger.w("Either conversationId or userId should be provided to SecurityClassificationViewModel")
```

Comment:
Maybe we should throw an exception if we don't have both conversationId and userId, because it's an illegal state

Reply:
I think warning is a better option because we are using it more for federation purposes and we may not catch exception before release 🤔

Reply:
Ok, I guess it's not a big deal, because if both are null then this banner simply won't be visible, right? It's just that throwing an exception in that case would let us know that someone screwed up in the code faster 😄 
But it was just a suggestion 😉